stdlib/public/core/ManagedBuffer.swift

```diff
@@ -34,22 +34,14 @@ open class ManagedBuffer<Header, Element> {
       ManagedBuffer<Header, Element>) throws -> Header
   ) rethrows -> ManagedBuffer<Header, Element> {
 
-    let p = try ManagedBufferPointer<Header, Element>(
-      bufferClass: self,
-      minimumCapacity: minimumCapacity,
-      makingHeaderWith: { buffer, _ in
-        try factory(
-          unsafeDowncast(buffer, to: ManagedBuffer<Header, Element>.self))
-      })
-
-    return unsafeDowncast(p.buffer, to: ManagedBuffer<Header, Element>.self)
-  }
+    let p = Builtin.allocWithTailElems_1(
+         self,
+         minimumCapacity._builtinWordValue, Element.self)
 
-  /// Destroy the stored Header.
-  deinit {
-    ManagedBufferPointer(self).withUnsafeMutablePointerToHeader {
-      _ = $0.deinitialize()
-    }
+    let initHeaderVal = try factory(p)
+    p.headerAddress.initialize(to: initHeaderVal)
+    _fixLifetime(p) 
```

Comment:
I suppose there's a rare case where `p` is returned and then thrown away, then this function is inlined, then the optimizer decides to move the release up to right after the call to `headerAddress`. (Since it's a bare pointer, the optimizer doesn't obviously know that it depends on `p` staying alive.)